src/models/summarization/samo_t5_summarizer.py

```
@@ -0,0 +1,321 @@
+            }
+        }
+        
+        if config_path and os.path.exists(config_path):
+            try:
+                with open(config_path, 'r', encoding='utf-8') as f:
+                    config = yaml.safe_load(f)
+                # Merge with defaults
+                for key, value in default_config.items():
+                    if key not in config:
+                        config[key] = value
+                return config
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)